base/tuple.jl

```diff
@@ -105,7 +105,8 @@ julia> ntuple(i -> 2*i, 4)
 ```
 """
 function ntuple(f::F, n::Integer) where F
-    t = n <= 0  ? () :
+    (n >= 0) || throw(DomainError())
```

Comment:
I've benchmarked the early and the late `n` check variants with
```julia
@benchmark ntuple(identity, n) setup=(n = rand(0:10))
```
The late one (inside `_ntuple()`) was 1ns (~10%) faster, so I've updated the PR.